### ISSUE TITLE
allowImportingTsExtensions not working with transpileOnly

### ISSUE THREAD

**Issue by lmiller1990** [TypeStrong/ts-node#1981]
### Search Terms

transpileOnly, allowImportingTsExtensions, typescript 5

Possibly related: https://github.com/TypeStrong/ts-node/issues/1978

### Expected Behavior

`allowImportingTsExtensions` working with `transpileOnly`

### Actual Behavior

Should work (I think?)

### Steps to reproduce the problem

- ✅ Works: `NODE_OPTIONS="--loader ts-node/esm" node index.ts`
- ❌ Does not work: `NODE_OPTIONS="--loader ts-node/esm/transpile-only" node index.ts`

### Minimal reproduction

https://github.com/lmiller1990/ts-node-transpile-error

### Specifications

ts-node v10.9.1
node v16.19.1
compiler v5.0.2

```
{
  "compilerOptions": {
    "allowImportingTsExtensions": true,
    "module": "ES2015",
    "noEmit": true
  }
}


```
* package.json:
```
{
  "type": "module",
  "name": "tsnode",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "ts-node": "^10.9.1",
    "typescript": "^5.0.2"
  }
}

```
* Operating system and version: macos ventura

* If Windows, are you using WSL or WSL2?:

**Comment by andrewbranch** [comment]
Should be fixed by https://github.com/microsoft/TypeScript/pull/53599

**Comment by lmiller1990** [comment]
Yep, let's wait for TS 5.1 and verify.

**Comment by andrewbranch** [comment]
Yes, it’s the same, and both will be fixed in a patch release 5.0.4 ~this week.

**Comment by lmiller1990** [comment]
Yep, it's fixed in 5.0.4 which is now out, thanks!
